storagemarket/deal_handler.go

```diff
@@ -37,3 +36,27 @@ func (d *dealHandler) subscribeUpdates() (event.Subscription, error) {
 	}
 	return sub, nil
 }
+
+func (dh *dealHandler) cancel() error {
+	dh.transferMu.Lock()
+	defer dh.transferMu.Unlock()
+
+	if dh.transferFinished {
+		return dh.transferErr
+	}
+
+	dh.transferCancel()
+
+	select {
+	case err := <-dh.transferDone:
+		dh.transferFinished = true
+		dh.transferErr = err
+		return err
+	case <-dh.providerCtx.Done():
+		return dh.providerCtx.Err()
+	}
+}
+
+func (dh *dealHandler) close() {
```

Comment:
nit: we don't really need this close method, you can just rename `transferCancel` to `close`

Reply:
@dirkmc This method will be called whenever a deal is cleaned up just so we can ensure all resources associated with a deal handler are reclaimed. I think it's good to keep this method and encapsulate all cleanup inside it rather then exposing the context cancellation to all the cleanup callsites.

Reply:
Ok makes sense